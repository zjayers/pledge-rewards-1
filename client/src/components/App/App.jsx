/* Import Modules */
import React from 'react';

/* Import Theme Provider */
import Theme from '../Theme';
import { GlobalStyles } from './App.style';

/* Import Components */
import RewardViewer from '../RewardViewer/RewardViewer';

/**
 * App Component
 * @returns {*} JSX to be rendered to the page
 * @constructor
 */
const App = () => {
  return (
    <Theme>
      <GlobalStyles />
      <RewardViewer />
    </Theme>
  );
};

/* Export this module */
export default App;
