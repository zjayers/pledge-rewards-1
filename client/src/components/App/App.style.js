import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Muli');
  * {
    font-family: 'Muli', sans-serif;
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  
  html {
    font-size: 62.5%;
    font-weight: 400;
  }
`;
