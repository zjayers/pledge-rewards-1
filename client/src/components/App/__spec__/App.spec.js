import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('App Component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<App />);
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });
});
