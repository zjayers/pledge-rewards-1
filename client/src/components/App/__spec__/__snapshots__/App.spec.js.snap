// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should contain the global style 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="5.1.1"
  >
    .BmWDh{width:100%;border:1px solid #9B9E9E;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;padding:20px 20px;margin-bottom:20px;-webkit-transition:all 0.2s ease-in-out;transition:all 0.2s ease-in-out;position:relative;}
    .iErRtB{font-size:1.8rem;color:#282828;font-weight:400;line-height:1.4;margin-bottom:5px;}
    .eiBTbF{font-size:1rem;color:#888B8B;text-transform:capitalize;}
    .cDJTT{font-size:1rem;color:#888B8B;text-transform:uppercase;}
    .cCvtGG{position:relative;margin:10px 0;}
    .evsVHR{-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;text-align:center;overflow:hidden;font-size:1.2rem;white-space:nowrap;width:4.2rem;}
    .jTmnGB{-webkit-transition:all 300ms cubic-bezier(0.175,0.885,0.335,1);transition:all 300ms cubic-bezier(0.175,0.885,0.335,1);height:100%;position:absolute;border-right:1px solid #dcdedd;top:0;left:0;display:-webkit-inline-box;display:-webkit-inline-flex;display:-ms-inline-flexbox;display:inline-flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;pointer-events:none;color:#656969;z-index:2;}
    .kOuXyP{border:1px solid #dcdedd;color:#282828;cursor:default;padding:10px 10px 10px 5.4rem;position:relative;height:42px;line-height:20px;box-sizing:border-box;width:100%;-webkit-transition:all 300ms cubic-bezier(0.175,0.885,0.335,1);transition:all 300ms cubic-bezier(0.175,0.885,0.335,1);text-rendering:auto;word-spacing:normal;text-transform:none;text-indent:0px;text-shadow:none;display:inline-block;text-align:start;font:400 13.3333px Arial;-webkit-writing-mode:horizontal-tb;-ms-writing-mode:lr;writing-mode:horizontal-tb;}
    .kOuXyP:focus{outline:0;border-color:#009e74;box-shadow:#f8f9fd 0 0 5px;}
    .kOuXyP:focus + div{border-color:#009e74;}
    .hTA-dXx{width:100%;background-color:#009e74;color:#ffffff;height:4.2rem;padding:0 1.8rem;font-size:1.4rem;font-weight:500;-webkit-transition:all 0.25s ease-in-out !important;transition:all 0.25s ease-in-out !important;text-align:center;cursor:pointer;border:none;position:relative;display:inline-block;border-radius:0;margin-top:18px;}
    .hTA-dXx:hover:not(:disabled){background-color:#037362;}
    .hTA-dXx:hover:not(:disabled){-webkit-transform:translateY(-1px);-ms-transform:translateY(-1px);transform:translateY(-1px);}
    .hTA-dXx:focus{outline:1px solid #282828;outline-offset:2px;-webkit-transition:outline 0.25s ease-in-out;transition:outline 0.25s ease-in-out;}
    .jYPjsU{height:100%;overflow:auto;padding-right:20px;padding-bottom:60px;padding-top:40px;}
    .dAlLHb{margin-top:10px;padding:1.8rem;width:100%;height:auto;background:linear-gradient(to bottom,#dbe7ff,#fff2ec);border:1px solid #dbe7ff;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}
    .dAlLHb span,.dAlLHb p{margin-bottom:1.2rem;font-size:1.4rem;line-height:2.4rem;}
    .dAlLHb span{font-weight:800;color:#222222;}
    .iORtcV{font-size:2.1rem;color:#282828;text-transform:capitalize;font-weight:400;line-height:1.4;margin:4rem 0;}
    *{font-family:'Muli',sans-serif;margin:0;padding:0;box-sizing:border-box;}
    html{font-size:62.5%;font-weight:400;}
    body{padding:20px;}
    *{font-family:'Muli',sans-serif;margin:0;padding:0;box-sizing:border-box;}
    html{font-size:62.5%;font-weight:400;}
    body{padding:20px;}
  </style>
</head>
`;

exports[`App Component should match the test snapshot 1`] = `
.c2 {
  width: 100%;
  border: 1px solid #9B9E9E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px 20px;
  margin-bottom: 20px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  position: relative;
}

.c3 {
  font-size: 1.8rem;
  color: #282828;
  font-weight: 400;
  line-height: 1.4;
  margin-bottom: 5px;
}

.c4 {
  font-size: 1rem;
  color: #888B8B;
  text-transform: capitalize;
}

.c9 {
  font-size: 1rem;
  color: #888B8B;
  text-transform: uppercase;
}

.c5 {
  position: relative;
  margin: 10px 0;
}

.c8 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  overflow: hidden;
  font-size: 1.2rem;
  white-space: nowrap;
  width: 4.2rem;
}

.c7 {
  -webkit-transition: all 300ms cubic-bezier(0.175,0.885,0.335,1);
  transition: all 300ms cubic-bezier(0.175,0.885,0.335,1);
  height: 100%;
  position: absolute;
  border-right: 1px solid #dcdedd;
  top: 0;
  left: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  color: #656969;
  z-index: 2;
}

.c6 {
  border: 1px solid #dcdedd;
  color: #282828;
  cursor: default;
  padding: 10px 10px 10px 5.4rem;
  position: relative;
  height: 42px;
  line-height: 20px;
  box-sizing: border-box;
  width: 100%;
  -webkit-transition: all 300ms cubic-bezier(0.175,0.885,0.335,1);
  transition: all 300ms cubic-bezier(0.175,0.885,0.335,1);
  text-rendering: auto;
  word-spacing: normal;
  text-transform: none;
  text-indent: 0px;
  text-shadow: none;
  display: inline-block;
  text-align: start;
  font: 400 13.3333px Arial;
  -webkit-writing-mode: horizontal-tb;
  -ms-writing-mode: lr;
  writing-mode: horizontal-tb;
}

.c6:focus {
  outline: 0;
  border-color: #009e74;
  box-shadow: #f8f9fd 0 0 5px;
}

.c6:focus + div {
  border-color: #009e74;
}

.c11 {
  width: 100%;
  background-color: #009e74;
  color: #ffffff;
  height: 4.2rem;
  padding: 0 1.8rem;
  font-size: 1.4rem;
  font-weight: 500;
  -webkit-transition: all 0.25s ease-in-out !important;
  transition: all 0.25s ease-in-out !important;
  text-align: center;
  cursor: pointer;
  border: none;
  position: relative;
  display: inline-block;
  border-radius: 0;
  margin-top: 18px;
}

.c11:hover:not(:disabled) {
  background-color: #037362;
}

.c11:hover:not(:disabled) {
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c11:focus {
  outline: 1px solid #282828;
  outline-offset: 2px;
  -webkit-transition: outline 0.25s ease-in-out;
  transition: outline 0.25s ease-in-out;
}

.c0 {
  height: 100%;
  overflow: auto;
  padding-right: 20px;
  padding-bottom: 60px;
  padding-top: 40px;
}

.c10 {
  margin-top: 10px;
  padding: 1.8rem;
  width: 100%;
  height: auto;
  background: linear-gradient(to bottom,#dbe7ff,#fff2ec);
  border: 1px solid #dbe7ff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 span,
.c10 p {
  margin-bottom: 1.2rem;
  font-size: 1.4rem;
  line-height: 2.4rem;
}

.c10 span {
  font-weight: 800;
  color: #222222;
}

.c1 {
  font-size: 2.1rem;
  color: #282828;
  text-transform: capitalize;
  font-weight: 400;
  line-height: 1.4;
  margin: 4rem 0;
}

<App>
  <Theme>
    <ThemeProvider
      theme={
        Object {
          "colors": Object {
            "borderGrey": "#9B9E9E",
            "buttonGreen": "#009E74",
            "flavorText": "#888B8B",
            "gradientBlue": "#DBE7FE",
            "gradientOrange": "#FFF2EC",
            "highlightGreen": "#009E74",
            "linkText": "#498F80",
            "text": "#282828",
          },
          "fontSizes": Object {
            "large": "3rem",
            "medium": "1.8rem",
            "small": "1rem",
          },
          "fonts": Array [
            "Muli",
            "sans-serif",
          ],
        }
      }
    >
      <HelmetWrapper
        defer={true}
        encodeSpecialCharacters={true}
      >
        <SideEffect(NullComponent)
          defer={true}
          encodeSpecialCharacters={true}
          link={
            Array [
              Object {
                "href": "https://fonts.googleapis.com/css2?family=Muli:ital,wght@0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap",
                "rel": "stylesheet",
              },
            ]
          }
          meta={
            Array [
              Object {
                "charset": "UTF-8",
              },
              Object {
                "content": "width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0",
                "name": "viewport",
              },
              Object {
                "content": "ie=edge",
                "http-equiv": "X-UA-Compatible",
              },
            ]
          }
          script={
            Array [
              Object {
                "crossOrigin": "anonymous",
                "src": "https://kit.fontawesome.com/1e96e77bac.js",
              },
            ]
          }
          title="Document"
          titleAttributes={Object {}}
        >
          <NullComponent
            defer={true}
            encodeSpecialCharacters={true}
            link={
              Array [
                Object {
                  "href": "https://fonts.googleapis.com/css2?family=Muli:ital,wght@0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap",
                  "rel": "stylesheet",
                },
              ]
            }
            meta={
              Array [
                Object {
                  "charset": "UTF-8",
                },
                Object {
                  "content": "width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0",
                  "name": "viewport",
                },
                Object {
                  "content": "ie=edge",
                  "http-equiv": "X-UA-Compatible",
                },
              ]
            }
            script={
              Array [
                Object {
                  "crossOrigin": "anonymous",
                  "src": "https://kit.fontawesome.com/1e96e77bac.js",
                },
              ]
            }
            title="Document"
            titleAttributes={Object {}}
          />
        </SideEffect(NullComponent)>
      </HelmetWrapper>
      <Memo(GlobalStyleComponent) />
      <RewardViewer
        getRewards={[Function]}
      >
        <RewardViewerstyle___default>
          <div
            className="c0"
          >
            <_default>
              <LargeTitlestyle___default>
                <h3
                  className="c1"
                >
                  Support
                </h3>
              </LargeTitlestyle___default>
            </_default>
            <GradientCard>
              <_default
                activated={true}
              >
                <Containerstyle___default
                  activated={true}
                >
                  <div
                    className="c2"
                  >
                    <_default>
                      <Headingstyle___default>
                        <h2
                          className="c3"
                        >
                          Pledge without a reward
                        </h2>
                      </Headingstyle___default>
                    </_default>
                    <WithTheme(Component)
                      activated={true}
                    >
                      <Component
                        activated={true}
                        theme={
                          Object {
                            "colors": Object {
                              "borderGrey": "#9B9E9E",
                              "buttonGreen": "#009E74",
                              "flavorText": "#888B8B",
                              "gradientBlue": "#DBE7FE",
                              "gradientOrange": "#FFF2EC",
                              "highlightGreen": "#009E74",
                              "linkText": "#498F80",
                              "text": "#282828",
                            },
                            "fontSizes": Object {
                              "large": "3rem",
                              "medium": "1.8rem",
                              "small": "1rem",
                            },
                            "fonts": Array [
                              "Muli",
                              "sans-serif",
                            ],
                          }
                        }
                      >
                        <_default
                          style={
                            Object {
                              "color": "#282828",
                            }
                          }
                          uppercase={false}
                        >
                          <SubHeadingstyle___default
                            uppercase={false}
                          >
                            <p
                              className="c4"
                            >
                              Pledge amount
                            </p>
                          </SubHeadingstyle___default>
                        </_default>
                        <CurrencyEntry
                          input=""
                          setInput={[Function]}
                        >
                          <CurrencyEntrystyle__Form>
                            <div
                              className="c5"
                            >
                              <CurrencyEntrystyle__Input
                                min="0"
                                onChange={[Function]}
                                type="number"
                                value=""
                              >
                                <input
                                  className="c6"
                                  min="0"
                                  onChange={[Function]}
                                  type="number"
                                  value=""
                                />
                              </CurrencyEntrystyle__Input>
                              <CurrencyEntrystyle__CurrencyBox>
                                <div
                                  className="c7"
                                >
                                  <CurrencyEntrystyle__CurrencyIcon>
                                    <span
                                      className="c8"
                                    >
                                      $
                                    </span>
                                  </CurrencyEntrystyle__CurrencyIcon>
                                </div>
                              </CurrencyEntrystyle__CurrencyBox>
                            </div>
                          </CurrencyEntrystyle__Form>
                        </CurrencyEntry>
                        <_default
                          uppercase={true}
                        >
                          <SubHeadingstyle___default
                            uppercase={true}
                          >
                            <p
                              className="c9"
                            >
                              About £0
                            </p>
                          </SubHeadingstyle___default>
                        </_default>
                        <GradientCardstyle___default>
                          <div
                            className="c10"
                          >
                            <span>
                              Back it because you believe in it.
                            </span>
                            <p>
                              Support the project for no rewards, just because it speaks to you.
                            </p>
                          </div>
                        </GradientCardstyle___default>
                        <ContinueButton>
                          <ContinueButtonstyle___default>
                            <button
                              className="c11"
                            >
                              Continue
                            </button>
                          </ContinueButtonstyle___default>
                        </ContinueButton>
                      </Component>
                    </WithTheme(Component)>
                  </div>
                </Containerstyle___default>
              </_default>
            </GradientCard>
          </div>
        </RewardViewerstyle___default>
      </RewardViewer>
    </ThemeProvider>
  </Theme>
</App>
`;
